bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/form/DateTextField.java

```diff
@@ -27,6 +37,104 @@
 public class DateTextField extends org.apache.wicket.extensions.markup.html.form.DateTextField {
     private static final long serialVersionUID = 3499287675713818823L;
 
+    private static final String EVENT_PARAM = "datePickerEvent";
+    private static final String DATE = "date";
+
+    /**
+     * Enumeration for date picker client side events.
+     */
+    public enum Event {
+        show, // Fired when the date picker is displayed.
+        hide, // Fired when the date picker is hidden.
+        clearDate, // Fired when the date is cleared, normally when the “clear” button (enabled with the clearBtn option) is pressed.
+        changeDate, // Fired when the date is changed.
+        changeMonth, // Fired when the view month is changed from year view.
+        changeYear, // Fired when the view year is changed from decade view.
+        changeDecade, // Fired when the view decade is changed from century view.
+        changeCentury, // Fired when the view century is changed from millennium view.
+    }
+
+    /**
+     * Abstract base class for events.
+     */
+    public abstract class AbstractEventHandler implements Serializable {
+
+        protected abstract String getBody();
+
+        protected String getFunction() {
+            StringBuilder sb = new StringBuilder("function(e) {\n");
+            sb.append(getBody());
+            sb.append("\n}");
+            return sb.toString();
+        }
+    }
+
+    /**
+     * Ajax event handler
+     */
+    public interface IAjaxEvenHandler extends Serializable {
+
+        void onAjaxEvent(AjaxRequestTarget target, Date date);
+
+    }
+
+    private abstract class AbstractAjaxEvent extends AbstractEventHandler {
+
+        private final Event event;
+        private final IAjaxEvenHandler handler;
+
+        public AbstractAjaxEvent(Event event, IAjaxEvenHandler handler) {
+            this.event = event;
+            this.handler = handler;
+        }
+
+        protected void updateAjaxAttributes(AjaxRequestAttributes attributes) {
+            attributes.getExtraParameters().put(EVENT_PARAM, event.name());
+            attributes.getExtraParameters().put(DATE, new JsonFunction("e.format()"));
+        }
+
+        protected void onAjaxEvent(AjaxRequestTarget target) {
+            String dateStr = RequestCycle.get().getRequest().getRequestParameters().getParameterValue(DATE).toString(null);
+            Date date = !Strings.isEmpty(dateStr)?
+                DateTextField.this.getConverter(Date.class).convertToObject(dateStr, RequestCycle.get().getRequest().getLocale()): null;
+            DateTextField.this.setDefaultModelObject(date);
```

Comment:
@martin-g 

For my use case on date selected it is convenient to have the model value set... but it might be the case user do not want it set. Make it configurable via a param to addHandler?